app/screens/sso/sso_authentication_with_external_browser.tsx

```diff
@@ -0,0 +1,166 @@
+            textAlign: 'center',
+            ...typography('Body', 200, 'Regular'),
+        },
+        infoContainer: {
+            alignItems: 'center',
+            flex: 1,
+            justifyContent: 'center',
+        },
+        infoText: {
+            color: changeOpacity(theme.centerChannelColor, 0.72),
+            ...typography('Body', 100, 'Regular'),
+        },
+        infoTitle: {
+            color: theme.centerChannelColor,
+            marginBottom: 4,
+            ...typography('Heading', 700),
+        },
+    };
+});
+
+const SSOAuthenticationWithExternalBrowser = ({doSSOLogin, loginError, loginUrl, serverUrl, setLoginError, theme}: SSOWithRedirectURLProps) => {
+    const [error, setError] = useState<string>('');
+    const [loginSuccess, setLoginSuccess] = useState(false);
+    const style = getStyleSheet(theme);
+    const intl = useIntl();
+    let customUrlScheme = Sso.REDIRECT_URL_SCHEME;
+    if (isBetaApp) {
+        customUrlScheme = Sso.REDIRECT_URL_SCHEME_DEV;
+    }
+
+    const redirectUrl = customUrlScheme + 'callback';
+    const init = (resetErrors = true) => {
+        setLoginSuccess(false);
+        if (resetErrors !== false) {
+            setError('');
+            setLoginError('');
+            NetworkManager.invalidateClient(serverUrl);
+            NetworkManager.createClient(serverUrl);
+        }
+        const parsedUrl = urlParse(loginUrl, true);
+        const query: Record<string, string> = {
+            ...parsedUrl.query,
+            redirect_to: redirectUrl,
+        };
+        parsedUrl.set('query', qs.stringify(query));
+        const url = parsedUrl.toString();
+
+        const onError = (e: Error) => {
+            let message;
+            if (e && Platform.OS === 'android' && isErrorWithMessage(e) && e.message.match(/no activity found to handle intent/i)) {
+                message = intl.formatMessage({
+                    id: 'mobile.oauth.failed_to_open_link_no_browser',
+                    defaultMessage: 'The link failed to open. Please verify that a browser is installed on the device.',
+                });
+            } else {
+                message = intl.formatMessage({
+                    id: 'mobile.oauth.failed_to_open_link',
+                    defaultMessage: 'The link failed to open. Please try again.',
+                });
+            }
+            setError(
+                message,
+            );
+        };
+
+        tryOpenURL(url, onError);
+    };
+
+    useEffect(() => {
+        const onURLChange = ({url}: { url: string }) => {
+            if (url && url.startsWith(redirectUrl)) {
+                const parsedUrl = urlParse(url, true);
+                const bearerToken = parsedUrl.query?.MMAUTHTOKEN;
+                const csrfToken = parsedUrl.query?.MMCSRF;
+                if (bearerToken && csrfToken) {
+                    setLoginSuccess(true);
+                    doSSOLogin(bearerToken, csrfToken);
```

Comment:
I guess we don't expect this prop to change, or at least not in a meaningful way, but if we don, we should rethink the dependencies.

The same for all the props use by `init`, since this would only take the `init` generated on the first run.

Reply:
If is not broken dont fix it.